client/modules/users/components/SignUp.js

```diff
@@ -28,11 +28,16 @@ class SignUp extends React.Component {
     this.props.clearFlags()
   }
 
-  onFieldChange = e => {
+  setFieldAsTouched = e => {
     const {name, value} = e.target
     this.setState({[name]: {value, touched: true}})
   }
 
+  setFieldValue = e => {
+    const {name, value} = e.target
+    this.setState({[name]: {value}})
```

Comment:
this clears the touched state when the value is changed, so the errors disappear even if the new value is still invalid

Reply:
I did that due to the adding of the onBlur event you asked. The validation should be triggered on the onBlur event, not while typing. And, due to the validation displays or hides the error reactively I thought that is better like this way.

Reply:
The usual way is the first validation is on blur, so you don't get an error while first filling out the field. Then validation happens on every change, so you can see when it's valid without having to leave the field.